sample-app/src/main/java/com/joinforage/android/example/pos/receipts/templates/txs/SnapPurchaseTxReceipt.kt

```diff
@@ -11,8 +11,8 @@ internal class SnapPurchaseTxReceipt : TxReceiptTemplate {
         merchant: Merchant?,
         terminalId: String,
         paymentMethod: PosPaymentMethod?,
-        payment: PosPaymentResponse
-    ) : super(merchant, terminalId, paymentMethod, payment.receipt!!)
+        receipt: Receipt
+    ) : super(merchant, terminalId, paymentMethod, receipt)
```

Comment:
Given that the constructor accepts a `Receipt` instead of the `PosPaymentResponse`, is the second constructor even needed anymore? I only made two constructors before so that it was easy to handle both `Payment`s and `Response`s but if we're just passing the receipt directly, it's unclear whether the second constructor has a point anymore 🤷

Reply:
Yeah I thought about this too. Seems like we might be able to reduce it down to just one if we stick with this approach (imo the sdk/api should return a receipt in an error scenario too so this whole thing becomes unnecessary). 

I liked having a constructor for each, but for these decline receipts it was a lot easier to only need to create a manual receipt instead of having to create an entire payment or refund.